api/v1/cluster_webhook.go

```diff
@@ -904,6 +908,122 @@ func (r *Cluster) validateConfiguration() field.ErrorList {
 	return result
 }
 
+// validateWalSizeConfiguration verifies that min_wal_size < max_wal_size < wal volume size
+func validateWalSizeConfiguration(
+	postgresConfig PostgresConfiguration, walVolumeSize *resource.Quantity,
+) field.ErrorList {
+	const (
+		minWalSizeKey     = "min_wal_size"
+		minWalSizeDefault = "80MB"
+		maxWalSizeKey     = "max_wal_size"
+		maxWalSizeDefault = "1GB"
+	)
+
+	var result field.ErrorList
+
+	minWalSize, hasMinWalSize := postgresConfig.Parameters[minWalSizeKey]
+	if minWalSize == "" {
+		minWalSize = minWalSizeDefault
+		hasMinWalSize = false
+	}
+	minWalSizeValue, err := parseWalSettingValue(minWalSize)
+	if err != nil {
+		result = append(
+			result,
+			field.Invalid(
+				field.NewPath("spec", "postgresql", "parameters", minWalSizeKey),
+				minWalSize,
+				fmt.Sprintf("Invalid value for configuration parameter %s", minWalSizeKey)))
+	}
+
+	maxWalSize, hasMaxWalSize := postgresConfig.Parameters[maxWalSizeKey]
+	if maxWalSize == "" {
+		maxWalSize = maxWalSizeDefault
+		hasMaxWalSize = false
+	}
+	maxWalSizeValue, err := parseWalSettingValue(maxWalSize)
+	if err != nil {
+		result = append(
+			result,
+			field.Invalid(
+				field.NewPath("spec", "postgresql", "parameters", maxWalSizeKey),
+				maxWalSize,
+				fmt.Sprintf("Invalid value for configuration parameter %s", maxWalSizeKey)))
+	}
+
+	if !minWalSizeValue.IsZero() && !maxWalSizeValue.IsZero() &&
+		minWalSizeValue.Cmp(maxWalSizeValue) >= 0 {
+		result = append(
+			result,
+			field.Invalid(
+				field.NewPath("spec", "postgresql", "parameters", minWalSizeKey),
+				minWalSize,
+				fmt.Sprintf("Invalid configuration. Parameter %s (default %s) should be smaller than parameter %s (default %s)",
+					minWalSizeKey, minWalSizeDefault, maxWalSizeKey, maxWalSizeDefault)))
+	}
+
+	if walVolumeSize == nil {
+		return result
+	}
+
+	if hasMinWalSize &&
+		!minWalSizeValue.IsZero() &&
+		minWalSizeValue.Cmp(*walVolumeSize) >= 0 {
+		result = append(
+			result,
+			field.Invalid(
+				field.NewPath("spec", "postgresql", "parameters", minWalSizeKey),
+				minWalSize,
+				fmt.Sprintf("Invalid configuration. Parameter %s (default %s) should be smaller than WAL volume size",
+					minWalSizeKey, minWalSizeDefault)))
+	}
+
+	if hasMaxWalSize &&
+		!maxWalSizeValue.IsZero() &&
+		maxWalSizeValue.Cmp(*walVolumeSize) >= 0 {
```

Comment:
should remove = in compare `maxWalSizeValue.Cmp(*walVolumeSize) > 0`, so the default storageSize 1Gi is not broken ?

Reply:
I considered this, but that will give the user a false sense of correctness. Given that it is not a hard limit in Postgres space usage, setting the maximum equal to the volume size is not correct.